src/software/backend/simulation/simulator_ball.cpp

```diff
@@ -0,0 +1,50 @@
+#include "software/backend/simulation/simulator_ball.h"
+
+std::weak_ptr<PhysicsBall> SimulatorBall::physics_ball_weak_ptr =
+    std::weak_ptr<PhysicsBall>();
```

Comment:
Is this necessary since it's already been declared in the header? Is this being redeclared and hiding the variable in the header by the same name?

Reply:
Because it's a static variable it needs to be instantiated in a `.cpp` file. Even in the header it doesn't get default constructed without being explicitly initialized like this. (it is not hiding, it is initializing).

Reply:
Are we making this static so that we can get it's methods for passing into firmware? Could we wrap the methods in the same way that we "wrap" the robot methods so that we don't need static variables?

If we don't end up doing the above, we should at least have the constructor throw an exception if we try to create more then one instance of this class.

Reply:
> Are we making this static so that we can get it's methods for passing into firmware? Could we wrap the methods in the same way that we "wrap" the robot methods so that we don't need static variables?

It's static mostly because everything else in the singleton class is, and I'm assuming the user doesn't need to actually create an instance of the singleton. It's more clear if it's all static.

It's used in the implementation of each static function that will be bound to firmware. The robot class also uses static variables this way, so I don't understand what you mean by "wrapping" again.

> If we don't end up doing the above, we should at least have the constructor throw an exception if we try to create more then one instance of this class.

This is also a "singleton" controller class. It doesn't get instantiated / created. It's totally legal to call `setPhysicsBall` multiple times if we ever wanted to change what ball is being used / controlled.

Reply:
It doesn't look like an initialization cuz the type is being re-declared but I'm ok with it if that's how we want to do singletons.

Reply:
Yeah, after looking at this again, I'm fine with this as-is.